### PR TITLE
Charging station   list of charging plans

### PR DESCRIPTION
**Pull request by OliveGerste**

### CODE REVIEW

File: src/app/pages/charging-stations/charging-limit/charging-profile-limit/charging-station-charging-profile-limit.component.ts [hunk 1]
```diff
@@ -44,10 +44,14 @@ export class ChargingStationChargingProfileLimitComponent implements OnInit, Aft
   @Input() public chargingStation!: ChargingStation;
 
   public profileTypeMap: ProfileType[] = [
-    { key: ChargingProfileKindType.ABSOLUTE, description: 'chargers.smart_charging.profile_types.absolute',
-      chargingProfileKindType: ChargingProfileKindType.ABSOLUTE, stackLevel: 3, profileId: 3 },
-    { key: ChargingProfileKindType.RECURRING, recurrencyKindType: RecurrencyKindType.DAILY, description: 'chargers.smart_charging.profile_types.recurring_daily',
-      chargingProfileKindType: ChargingProfileKindType.RECURRING, stackLevel: 2, profileId: 2 },
+    {
+      key: ChargingProfileKindType.ABSOLUTE, description: 'chargers.smart_charging.profile_types.absolute',
+      chargingProfileKindType: ChargingProfileKindType.ABSOLUTE, stackLevel: 3, profileId: 3
```
Comment (mahdibenromdhane) [inline src/app/pages/charging-stations/charging-limit/charging-profile-limit/charging-station-charging-profile-limit.component.ts:49]: Are profileId and stack level static ? if yes , maybe it's better to use enum

File: src/app/pages/charging-stations/charging-limit/charging-profile-limit/charging-station-charging-profile-limit.component.ts [hunk 2]
```diff
@@ -44,10 +44,14 @@ export class ChargingStationChargingProfileLimitComponent implements OnInit, Aft
...
+    },
+    {
+      key: ChargingProfileKindType.RECURRING, recurrencyKindType: RecurrencyKindType.DAILY, description: 'chargers.smart_charging.profile_types.recurring_daily',
+      chargingProfileKindType: ChargingProfileKindType.RECURRING, stackLevel: 2, profileId: 2
```
Comment (mahdibenromdhane) [inline src/app/pages/charging-stations/charging-limit/charging-profile-limit/charging-station-charging-profile-limit.component.ts:53]: Same as above

File: src/app/pages/charging-stations/cell-components/charging-plans-current-limit-cell.component.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        const currentPeriod = this.row.profile.chargingSchedule.chargingSchedulePeriod.findIndex(p => p.startPeriod > currentDuration);
+        if (this.row.profile.chargingSchedule.chargingSchedulePeriod[currentPeriod - 1]) {
+          this.currentLimit = this.row.profile.chargingSchedule.chargingSchedulePeriod[currentPeriod - 1].limit;
+        } else {
+          this.currentLimit = this.row.profile.chargingSchedule.chargingSchedulePeriod
+          [this.row.profile.chargingSchedule.chargingSchedulePeriod.length - 1].limit;
+        }
+      } else {
+        if (this.row.connectorID === 0) {
+          this.currentLimit = Math.round(this.row.chargingStation.maximumPower / 230);
```
Comment (mahdibenromdhane) [inline src/app/pages/charging-stations/cell-components/charging-plans-current-limit-cell.component.ts:63]: We should use the VOLTAGE_230 variable in the Voltage enum

File: src/app/pages/charging-stations/cell-components/charging-plans-current-limit-cell.component.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      if (currentDuration > 0 && currentDuration < this.row.profile.chargingSchedule.duration) {
+        const currentPeriod = this.row.profile.chargingSchedule.chargingSchedulePeriod.findIndex(p => p.startPeriod > currentDuration);
+        if (this.row.profile.chargingSchedule.chargingSchedulePeriod[currentPeriod - 1]) {
+          this.currentLimit = this.row.profile.chargingSchedule.chargingSchedulePeriod[currentPeriod - 1].limit;
+        } else {
+          this.currentLimit = this.row.profile.chargingSchedule.chargingSchedulePeriod
+          [this.row.profile.chargingSchedule.chargingSchedulePeriod.length - 1].limit;
+        }
+      } else if (currentDuration < 0 && currentDuration + 86400 < this.row.profile.chargingSchedule.duration) {
+        currentDuration = 86400 + currentDuration;
```
Comment (mahdibenromdhane) [inline src/app/pages/charging-stations/cell-components/charging-plans-current-limit-cell.component.ts:53]: If 86400 static, we should use it globaly

**Comment by mahdibenromdhane** [comment]
image

I think the all property need to be removed( it does when we reset the filter).
Otherwise the UI seems to work fine.

File: src/app/pages/charging-stations/charging-limit/charging-profile-limit/charging-station-charging-profile-limit.component.ts (hunk 1 above)
Comment (OliveGerste) [reply src/app/pages/charging-stations/charging-limit/charging-profile-limit/charging-station-charging-profile-limit.component.ts:49]: I do not think, that it is going to be static in the future.

File: src/app/pages/charging-stations/charging-limit/charging-profile-limit/charging-station-charging-profile-limit.component.ts (hunk 2 above)
Comment (OliveGerste) [reply src/app/pages/charging-stations/charging-limit/charging-profile-limit/charging-station-charging-profile-limit.component.ts:53]: I do not think, that it is going to be static in the future.

**Comment by OliveGerste** [comment]
> image
> 
> I think the all property need to be removed( it does when we reset the filter).
> Otherwise the UI seems to work fine.

This has something to do with the table framework. Same happens in e.g. session table.

**Pull request merged**